### DISCUSSION TITLE
Meine Erfahrungen zum CaravanPi

### DISCUSSION THREAD

**Discussion by Casaluifred** [spitzlbergerj/CaravanPi#32] [Zeige und erzähle]
Eigentlich wollte ich selber etwas ähnliches auf Basis des Arduino Mega aufbauen, das Konzept vom Josef - vor allem auf Basis des MagicMirror hat mich dann aber überzeugt. Den MM habe ich mir für's Badezimmer aufgebaut. Diese Oberfläche auch für meinen Camper zu nutzen, finde ich genial.
Nun bin ich in Sachen Arduino eher so zwischen Anfänge rund Fortgeschrittener angesiedelt, der RaspBerry hat bei mir schon auch Einzug gehalten, aber eher mit ziemlich fertig verfügbaren Projekten.
Als Basis waren zwei Raspi 4 verfügbar, ein 10" Monitor mit HDMI und Touch (mal sehen, ob wir den auch noch nutzen können), Lötkolben, Kleinteile, ein 3D-Drucker, Lust auf das Projekt und natürlich zwei Camper (meiner und der vom Sohn).

Hab mir also eine Raspberry mit Bookworm installiert und mit der Installation des CampervanPi begonnen. Aufgrund schlechter Erfahrungen auch mit guten Micro-SD-Karten am Raspi habe ich das ganze auf eine m.2 Festplatte gepackt.

Zuerst habe ich es mit dem Script versucht, hat aber ständig mit Fehlern abgebrochen. 
Problem war hier, dass ich den Pi-Benutzer immer ändere (alte Admin-Krankheit ;) ).
Also habe ich es manuell versucht und den Pfad jeweils angepasst.
Hat aber auch nicht wirklich funktioniert, weshalb ich schlichtweg den Raspi neu aufgesetzt habe - diesmal mit Bullseye und den Benutzer Pi gelassen.
So hat dann das Script vom Josef funktioniert.
Kleiner Schönheitsfehler bei mir: Irgendwann scrollte der Bildschirm nicht mehr weiter, sondern alle Befehle wurden in der unteren Zeile übereinandergeschrieben - ungünstig, wenn genau dann die Abfragen kommen.
Einfache Abhilfe ist hier, wenn man im Script zwischendurch mal einen clr-Befehl einbaut. Kann aber auch sein, dass ich da ein Einzelfall bin.

Gut, der Raspi ist dann auch gestartet und hat die Magic-Mirror Oberfläche ebenfalls gestartet. Die habe ich dann per Putty-Kommandozeile erstmal wieder gestoppt, um auf den Dateiexplorer, Konsole und Editor zugreifen zu können.

Was nicht funktioniert hat, war die Einbindung der Bibliotheken für den Lagesensor und die Klimasensoren. 
Das wird in der Anleitung so beschrieben:
```
pip3 install adafruit-circuitpython-lis3dh
pip3 install adafruit-circuitpython-busdevice
pip3 install adafruit-circuitpython-adxl34x
```
Bei mir hat das immer wieder abgebrochen. Offenbar wurde die Bibliothek verändert.
Funktioniert hat es dann, als ich an jeden Befehl noch eine Erweiterung angehängt habe - hier zum Beispiel die zweite Zeile:
`pip3 install adafruit-circuitpython-busdevice --break-system-packages`

Als ich dann die Tests in der Beschreibung durchgeführt habe, bekam ich wieder Fehlermeldungen, dass er auf die Datei nicht zugreifen kann - z.B. position.
Auf dem Raspi geschaut, war die auch nicht da. Hab sie dann manuell angelegt und wieder getestet, dann hat er die zwar gefunden, aber keine Inhalte.
Letztendlich ist mir dann aufgefallen, dass eine Datei ADXL345 vorhanden ist und darin lagen dann die aktuellen Messwerte. Also die Sourcen auf den Dateinamen angepasst, dann hat das funktioniert.

In der Zwischenzeit wurden auch die beiden Platinen geliefert, die ich beim Josef bestellt habe.
Für eine der beiden habe ich dann die Wago-Leiterplattenklemmen bestellt und das Ganze dann zusammengelötet.
Hier könnte man die Bedruckung ein wenig nachbessern, denn aus alter Gewohnheit habe ich den 40poligen Wannenstecker so aufgelötet, dass die Aussparung für die Nase des Steckers nach außen zeigt. Das ist in dem Fall verkehrt Wenn man die Platine so hinlegt, dass man das Logo lesen kann, dann ist der Pin1 unten rechts. Hat man ein Flachbandkabel, bei dem der Pin1 farblich markiert ist, dann muss der Wannenstecker mit der Aussparung nach innen aufgelötet werden. 
Da man am Raspi selbst den Stecker in beiden Richtungen aufstecken kann, habe ich auf meinem Kabel nur beschriftet, dass es falsch eingebaut ist - wer will wegen sowas schon den Raspi schrotten ;)
Hier würde helfen, wenn man entweder die Bedruckung auf Wannenstecker ändert - wie das bei den anderen Wannensteckern auch gelöst ist - oder wenigstens, wenn man den Pin 1 markiert.

Wo ich auch erst nachmessen musste, ist die Belegung der Sensoren auf den Wago-Klemmen.
Hier fehlt noch ein Anschlussplan. Wenn ich fertig bin, dann stelle ich den aber gerne zur Verfügung.

Soweit erstmal mein aktueller Stand. Ich weiß nicht, ob ich der Einzige bin, der diese Probleme hatte, wenn nein, dann möchte ich damit anderen (Anfängern) helfen, das Ding zum Fliegen zu bekommen.
Der Josef hat da eine tolle Basis geschaffen, ich denke, wenn jeder mit seiner Erfahrung beiträgt, dann wird das eine Top-Lösung, die sich hinter kommerziellen Lösungen nicht verstecken braucht.

**Comment by spitzlbergerj** [comment]
Servus Fred,
vielen Dank, dass du hier Deine Erfahrungen teilst. So kann aus dem CaravanPi ein Community Projekt werden. Danke

Vielen Dank für den Hinweis mit dem User pi und die Abhängigkeit des Skripts (und vermutlich der Python Skripte) von diesem Namen. Daran hab ich noch gar nicht gedacht. Im Herbst arbeite ich wieder an der Software, dann schau ich mir das an.

Das Installieren von Libraies über pip wurde in der neuesten Raspberry Pi OS Version eingeschränkt. pip soll so nicht mehr verwendet werden. Leider hat adafruit darauf noch nicht reagiert und nutzt weiterhin pip zum Installieren. Das Skript sollte das schon berücksichtigen und mit der von dir genutzten Option installieren. Nach dem Durchlaufen des Skriptes solltest du also gar keine Lib mehr manuell installieren müssen. Die sollten alle da sein. Ein Blick in das Skript lohnt also immer.

Die Kommunikation der Sensor-Auslese-Programme mit den MagicMirror Modulen über die Dateien im Ordner values ist veraltet. Ich habe inzwischen alles auf die MariaDB umgestellt. Die Dateikommunikation ist nur noch aus Kompatibilitätsgründen da. Ich entwickle das aber nicht mehr weiter. Am MagicMirror sollte dann ein Grafana Modul genutzt werden, um die Daten anzuzeigen. Das wirkt auch erheblich moderner. Die datei-basierten MagicMirror Module werde ich auch nicht mehr weitere entwickeln.

Hier ein Beispiel für den neuen Aufbau der CaravanPi Anzeige. Alle Daten der Sensoren werden über Grafana angezeigt

![Screenshot 2024-08-03 202417](https://github.com/user-attachments/assets/c0cc6925-4f49-4d14-84b2-7bf482e3fe27)

Die SQL Statements für die Grafana-Abfragen kommen demnächst per Newsletter. Wenn du die vorher schon benötigst, melde dich bitte.

Die Wago Klemmen sind mir inzwischen zu teuer geworden. Und außerdem kann man da nicht schnell mal Geräte ausstecken, austauschen oder gar die Platine erneuern. Deswegen stelle ich bei mir gerade auf Stecker von JST um. Das zeige ich auch im neuen Newsletter im Herbst.

Die Platine V6 hat für den 40-poligen GPIO Stecker die 1 aufgedruckt. Sorry, das fiel mir schon lange auf, ich hab es aber immer wieder vergessen ins Platinendesign einzufügen.

Die Belegung der Wago-Klemmen bzw. der Kontaktflächen für die Sensoren ist aus dem [Schaltplan](https://raw.githubusercontent.com/spitzlbergerj/CaravanPi/master/circuit-board/V5/CaravanPi_Platine_V5_Schaltplan_Page_1.png) gut zu entnehmen. Ich meine, mehr braucht es da nicht. Die V6 ist dabei komplett kompatibel zur V5 bis auf die Spannungsversorgung des HX711, die auf 3V3 umgestellt wurde.

Viele Grüße
Sepp
